### PR TITLE
Fix jobscheduling

### PR DESCRIPTION
**Pull request by brycekbargar**
### TLDR;

We were using a bitmask stored as an integer in order to maintain the days of the week the job was scheduled to run. There was a lot of complicated math around shifting schedules and bits that led to a bug in scheduling for the same day.

We've now replaced the bitmask with an enum that has a FlagsAttribute. This allow for some simpler calculations around the next execution day/time as well as some more human readable code

### There be monsters
There is still a weird bug with scheduling for later today if you convert the JobSchedule to UTC before doing the next execution calculation. There wasn't enough time to actually fix it so instead I just ordered the code "correctly".

### CODE REVIEW

**Comment by brycekbargar** [comment]
The ui is broken and needs to be fixed : (

**Review by bgharper — APPROVED** [review]

**Comment by brycekbargar** [comment]
> The ui is broken and needs to be fixed : (

I just finished the refactor from int to enum and left the ExecutionSchedule property un-renamed.
